helix-term/src/commands/typed.rs

```diff
@@ -321,8 +321,26 @@ fn set_line_ending(
         arg if arg.starts_with("nel") => Nel,
         _ => bail!("invalid line ending"),
     };
+    let (view, doc) = current!(cx.editor);
+    doc.line_ending = line_ending;
+
+    let mut pos = 0;
+    let transaction = Transaction::change(
+        doc.text(),
+        doc.text().lines().filter_map(|line| {
+            pos += line.len_chars();
```

Comment:
Is it better to get the position of the line from `text` using `line` rather than manually counting it ourselves?

Reply:
I think that would be inefficient since it would be `O(log N)` time.

Reply:
`lines()` itself is `O(log N)`.

Reply:
I mean that `line_to_char` is `O(log N)`.